src/pandapipes/component_models/heat_consumer_component.py

```diff
@@ -170,10 +169,12 @@ def adaption_after_derivatives_hydraulic(cls, net, branch_pit, node_pit, idx_loo
             t_out = hc_pit[:, TOUTINIT]
 
             df_dm = - cp * (t_out - t_in)
-            hc_pit[mask, LOAD_VEC_BRANCHES] = - hc_pit[mask, QEXT] + df_dm[mask] * hc_pit[mask, MDOTINIT]
             mask_equal = t_out == t_in
-            hc_pit[mask & mask_equal, MDOTINIT] = 0
-            hc_pit[mask & ~mask_equal, JAC_DERIV_DM] = df_dm[mask & ~mask_equal]
+            mask_zero = hc_pit[:, QEXT] == 0
+            mask_ign = mask_equal | mask_zero
+            hc_pit[mask & mask_ign, MDOTINIT] = 0
```

Comment:
Is there a case where the derivative could "accidentally" be 0, while mdot is set to 0? I just fear that this could lead to problems with the solver.

Reply:
No, this would not happen, as the derivative is usually one. Look here:
![grafik](https://github.com/user-attachments/assets/e3b1ae48-d8ad-45ee-ae87-21bde78fd7a5)

Reply:
In all other cases except from the cases where Q and TR are defined JAC_DERIV_DM is equal to one.